src/blockdata/transaction.rs

```diff
@@ -238,22 +242,48 @@ impl BitcoinHash for Transaction {
     }
 }
 
-impl_consensus_encoding!(TxIn, prev_hash, prev_index, script_sig, sequence);
 impl_consensus_encoding!(TxOut, value, script_pubkey);
 
+impl<S: SimpleEncoder> ConsensusEncodable<S> for TxIn {
+    fn consensus_encode(&self, s: &mut S) -> Result <(), S::Error> {
+        try!(self.prev_hash.consensus_encode(s));
+        try!(self.prev_index.consensus_encode(s));
+        try!(self.script_sig.consensus_encode(s));
+        self.sequence.consensus_encode(s)
+    }
+}
+impl<D: SimpleDecoder> ConsensusDecodable<D> for TxIn {
+    fn consensus_decode(d: &mut D) -> Result<TxIn, D::Error> {
+        Ok(TxIn {
+            prev_hash: try!(ConsensusDecodable::consensus_decode(d)),
+            prev_index: try!(ConsensusDecodable::consensus_decode(d)),
+            script_sig: try!(ConsensusDecodable::consensus_decode(d)),
+            sequence: try!(ConsensusDecodable::consensus_decode(d)),
+            witness: vec![],
+        })
+    }
+}
+
 impl<S: SimpleEncoder> ConsensusEncodable<S> for Transaction {
     fn consensus_encode(&self, s: &mut S) -> Result <(), S::Error> {
         try!(self.version.consensus_encode(s));
-        if self.witness.is_empty() {
+        let mut have_witness = false;
+        for input in &self.input {
```

Comment:
You can do `let have_witness = self.input.iter().any(|i| !i.witness.is_empty());` and avoid the mutability, at the cost of a somewhat-crazy line of code. Up to you.

Reply:
Ugh, I hate pythonic-looking rust.

Reply:
Lol, k. Though FWIW I believe Python people would say "I hate haskell-looking python" to the same code.